components/feature/toolbar/src/main/java/mozilla/components/feature/toolbar/WebExtensionToolbarFeature.kt

```diff
@@ -94,6 +106,10 @@ class WebExtensionToolbarFeature(
     internal fun renderWebExtensionActions(state: BrowserState, tab: SessionState? = null) {
         val extensions = state.extensions.values.toList()
         extensions.filter { it.enabled }.sortedBy { it.name }.forEach { extension ->
+            if (isExtensionNotAllowedInPrivateBrowsing(extension, tab)) {
```

Comment:
We have a bug here. When long tapping on a link in a regular tab, selecting open in private tab and switching to it via the snackbar, we don't go through a start/stop cycle and would display actions of extensions that shouldn't run in private browsing. We can fix that though here using:

```kotlin
 if (isExtensionNotAllowedInPrivateBrowsing(extension, tab)) {
   webExtensionPageActions[extension.id]?.let { toolbar.removePageAction(it) }
   webExtensionBrowserActions[extension.id]?.let { toolbar.removeBrowserAction(it) }
   return@forEach
}
```